### PR TITLE
Miscellaneous bugfixes/syscall coverage improvement + small new feature

### PR DESCRIPTION
**Pull request by Keno**
As always, see commit messages for details. The first one `dump -t` is a small feature to filter dump entries by tid. The others are various bug fixes and missing syscall coverage I ran into recently. I probably have a few more of those coming, but wanted to get these in to avoid diverging too much from upstream. It's already a little large. If you want me to split up the PR, let me know.

### CODE REVIEW

**Comment by rocallahan** [comment]
The buildbot failed with a timeout in `timer`.

**Comment by Keno** [comment]
That buildbot really does not like me. I'll try to get around to debugging the failure soon.

File: src/DumpCommand.cc [hunk 1]
```diff
@@ -162,7 +172,8 @@ static void dump_events_matching(TraceReader& trace, const DumpFlags& flags,
     if (end < frame.time()) {
       return;
     }
-    if (start <= frame.time() && frame.time() <= end) {
+    if (start <= frame.time() && frame.time() <= end &&
+        (flags.only_tid == 0 || flags.only_tid == frame.tid())) {
```
Comment (rocallahan) [inline src/DumpCommand.cc]: `!flags.only_tid`

**Comment by Keno** [comment]
Ok. I believe I found it (see newest commit). Don't think it had anything to do with this PR, just unlucky ;).

**Comment by rocallahan** [comment]
Thanks again Keno!

**Pull request merged**